### ISSUE TITLE
Migrate from MvvmLight to Windows  Community Toolkit

### ISSUE THREAD

**Issue by ErikEJ** [ErikEJ/EFCorePowerTools#1294]
https://docs.microsoft.com/en-us/windows/communitytoolkit/mvvm/migratingfrommvvmlight

**Comment by ErikEJ** [comment]
@CorsairRO Are you still willing to attempt this?

**Comment by CorsairRO** [comment]
Hello,

sorry for that, i have been travelling a bit to some of our customers. This is why i've been not responding here.

I will investigate as soon as possible why the RaiseCanExecuteChanged doesnt work there which make the buttons inactive.

Again sorry for that, i have assumed that this should work there as i have been thinking of.

**Comment by ErikEJ** [comment]
I appreciate your effort, and there is no urgency to do this.

Let me know if you need help with testing if you take a second look, also, I have updated a number of dependencies in the latest master.

**Comment by CorsairRO** [comment]
No problem for my effort there, i am more than glad to be able to help.

It might just be some stupid thing in the new MVVM that starts all commands that have CanExecute with default of FALSE somehow. 

In my miniversion of your tool i have removed 90% of the features and just used 2 ContextMenu items in Visual Studio over the config.json(Edit and Refresh). This is why i didnt reproduce the issue there as in the first place. 
Assuming MVVM RelayCommand works as the previous.

**Comment by CorsairRO** [comment]
Hello again.

I have now corrected the issues in my stripped version ( Just FYI by my stripped version i mean i removed everything else then RevEng + Removed dacpac + Removed Schema VM/View + Removed ALL providers ( except SQL Server ) + 
Removed ALL options then EF Core 6 + Removed ALL Options ( Noda, MultipleResultsets, basically petty  much ALL tickboxes from options page +
Changed the CSharp Generator to generate the objects as i want ( being FULL properties instead of autogenerated, triggering PropertyChanged , all classes extend from Observable Base class and so on ... )

Will put the fixes in the branch that i got from you and check it in tomorrow most probably.

Xaml.Behaviors doesnt like anymore passing "true" or "false" from Xaml to a RelayCommand<bool>.
The error that i got earlier ( Specified cast is not valid )
Therefore i have changed ALL the bool commands in RevEng VM's to be RelayCommand<object>.

Also i had to add manually NotifyCanExecuteChanged in some places in order for the OK button to be active.

I will put the same fixes in other VM's as well ( checking for RelayCommand<bool> and trigerring NotifyCanExecuteChanges ) but i dont really use any other case then RevEng. 
i dont think i even know how to test some parts from there, like migrations and so on...

**Comment by ErikEJ** [comment]
I will test the PR when you make one

**Comment by CorsairRO** [comment]
Hello and again sorry for the long delay.

I will put now the PR stuff in 5 minutes. I am still testing it now after my changes there.

Basically main changes from prev attempt are:

1) ALL Commands that have RelayCommand<bool> are now converted to RelayCommand<object> and parsed accordingly.
I still dont know if it comes from Community MVVM or from Microsot.Xaml.Behaviors
2) Every VM that has OKCommand using a CanExecute should manually call to 
( (RelayCommand)OkCommand ).NotifyCanExecuteChanged( );

**Comment by CorsairRO** [comment]
Hello @ErikEJ 

It should be there now.
Did the PR gets pushed there? My VS doesnt say anything, and i am NOT used to this git stuff like you already know.
After seeing now how it works it will never work for my team. It is great maybe for collaborative teams all over world and also for github open source tools.

**Comment by ErikEJ** [comment]
@CorsairRO No it is not.

**Comment by CorsairRO** [comment]
@ErikEJ OK, i will try again now. I did a Push and stuff. I will try again, for sure i am doing something wrong there

**Comment by CorsairRO** [comment]
@ErikEJ I have tried another attempt to commit my changes to the WONDERFUL source control provider Git. Pleae let me know if this time is successful and you got any message there as i dont get anything back, just a message saying successful.
Last time it was saying the same and you aid you didnt get any change.

I have finished the changes there for 1 WK now. How hard can it be for a Source control provider to allow Push back some changes? ( not blaiming you with anything there, just joking about git )
